qiskit/visualization/matplotlib.py

```diff
@@ -119,6 +119,9 @@ def get_index(self):
 
 class MatplotlibDrawer:
     """Matplotlib drawer class called from circuit_drawer"""
+
+    _mathmode_regex = re.compile(r"(?<!\\)\$(.*)(?<!\\)\$")
```

Comment:
I like this alternative as class attribute. Would it be possible to in the other scenarios? Or there is performance penalty?

Reply:
From a performance perspective there's no real alternative between a class attribute and a global. I'll update the others to be class attrs